apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -585,6 +482,136 @@ impl Plugin for Telemetry {
 }
 
 impl Telemetry {
+    /// This method can be used instead of `Plugin::new` to override the subscriber
+    pub async fn new_with_subscriber<S>(
+        init: PluginInit<<Self as Plugin>::Config>,
+        subscriber: Option<S>,
+    ) -> Result<Self, BoxError>
+    where
+        S: Subscriber + Send + Sync + for<'span> LookupSpan<'span>,
+    {
+        // Apollo config is special because we enable tracing if some env variables are present.
+        let mut config = init.config;
+        let apollo = config
+            .apollo
+            .as_mut()
+            .expect("telemetry apollo config must be present");
```

Comment:
Should it be made non-optional in the config struct?

Reply:
it is optional in the configuration but will be modified by other means while initializing the router

Reply:
I mean it could be no-a-Rust-`Option` in the Rust struct, but have `#[serde(default)]` so it’s optional in YAML. But that doesn’t need to block this PR.